cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/KafkaCruiseControl.java

```diff
@@ -716,7 +724,8 @@ private boolean ignoreProposalCache(List<String> goals,
                               allowCapacityEstimation,
                               excludedTopics,
                               excludeRecentlyDemotedBrokers,
-                              excludeRecentlyRemovedBrokers);
+                              excludeRecentlyRemovedBrokers,
+                              isTriggeredByGoalViolation);
```

Comment:
If `isTriggeredByGoalViolation` is true, we need to ignore proposal cache since cached proposal always assumes [it is not triggered by anomaly detector.](https://github.com/linkedin/cruise-control/blob/97e8a66c320eec5a8767e7f75c6f95f5578edd76/cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/analyzer/GoalOptimizer.java#L346)